heron/stmgr/src/cpp/manager/stmgr-server.cpp

```diff
@@ -431,7 +433,15 @@ void StMgrServer::HandleTupleSetMessage(Connection* _conn,
         ->incr_by(_message->control().fails_size());
   }
   stmgr_->HandleInstanceData(iter->second, instance_info_[iter->second]->local_spout_, _message);
-  __global_protobuf_pool_release__(_message);
+  auto message_size = _message->ByteSize();
```

Comment:
ByteSizeLong?

Reply:
I could not find `ByteSizeLong` API in generated code. Let me try again...

Reply:
seems this is [introduced in 3.1.0](https://github.com/google/protobuf/blob/master/CHANGES.txt#L319) and then [deprecated in 3.4.0](https://github.com/google/protobuf/blob/master/CHANGES.txt#L58)

Reply:
also, ByteSize calculates size of serialized message, which seems will be slightly larger than actual size in memory.

Reply:
so if the message is 60MB but only little part of it is used to hold the incoming message (because incoming message is small), we will not delete it because I believe serialized size is much smaller than 60MB.